File: easy/Time_Conversion/cpp/time_conversion.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * Complete the timeConversion function below.
 */
string timeConversion(string s)
{
    /*
     * Write your code here.
     */
    string period_unit = s.substr(8, -1);
    string s_converted = s.substr(0, 8);
    string hh = s_converted.substr(0, 2);

    // https://en.wikipedia.org/wiki/12-hour_clock
    if ((hh.compare("12") != 0) && period_unit.compare("PM") == 0)
    {
        hh = to_string(stoi(s_converted) + 12);
    }
    else if ((hh.compare("12") == 0) && period_unit.compare("AM") == 0)
    {
        hh = "00";
    }

    s_converted = hh + s_converted.substr(2, -1);

    return s_converted;
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = timeConversion(s);

    fout << result << "\n";

    fout.close();

    return 0;
}

File: easy/Time_Conversion/python/time_conversion.py
#!/bin/python3

import os
import sys

#
# Complete the timeConversion function below.
#


def timeConversion(s):
    #
    # Write your code here.
    #
    period_unit = s[8:]
    s_converted = s[:8]
    hh = s_converted[:2]

    # https://en.wikipedia.org/wiki/12-hour_clock
    if hh != '12' and period_unit == 'PM':
        hh = str(int(hh) + 12)
    elif hh == '12' and period_unit == 'AM':
        hh = '00'

    s_converted = hh + s_converted[2:]

    return s_converted


if __name__ == '__main__':
    f = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = timeConversion(s)

    f.write(result + '\n')

    f.close()
